src/transformStateWithClones.js

```diff
@@ -6,8 +6,39 @@
  *
  * @return {Object[]}
  */
+
 function transformStateWithClones(state, actions) {
-  // write code here
+  const finalState = { ...state };
+  const stateHistory = [];
+
+  for (const order of actions) {
+    const ADD_PROPERTIES = order.type === 'addProperties';
+    const REMOVE_PROPERTIES = order.type === 'removeProperties';
+    const CLEAR_ALL = order.type === 'clear';
+
+    if (ADD_PROPERTIES) {
+      Object.assign(finalState, order.extraData);
+      stateHistory.push(Object.assign({}, finalState));
+    }
+
+    if (REMOVE_PROPERTIES) {
+      for (const remove of order.keysToRemove) {
+        delete finalState[remove];
+      }
+
+      stateHistory.push(Object.assign({}, finalState));
+    }
+
+    if (CLEAR_ALL) {
+      for (const key in finalState) {
+        delete finalState[key];
+      }
+
+      stateHistory.push(Object.assign({}, finalState));
+    }
+  }
```

Comment:
according to [checklist ](https://github.com/mate-academy/js_stateful-clones/blob/master/checklist.md) point 2 you should use switch statement